component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/SchemaImpl.java

```diff
@@ -308,7 +310,20 @@ public Builder swap(final String name, final String with) {
 
         @Override
         public Schema build() {
-            this.props.put(ENTRIES_ORDER_PROP, entriesOrder.stream().collect(Collectors.joining(",")));
+            this.props.put(ENTRIES_ORDER_PROP, entriesOrder.stream().collect(joining(",")));
+            return new SchemaImpl(this);
+        }
+
+        @Override
+        public Schema build(final EntriesOrder order) {
+            final List<String> existing = entries.stream().map(e -> e.getName()).collect(toList());
```

Comment:
This code should be enough
```java
            final String propEntriesOrder = entries.stream().sorted(order).map(Entry::getName).collect(joining(",");
            this.props.put(ENTRIES_ORDER_PROP, propEntriesOrder);
```
and `public Schema build(final EntriesOrder order) {` can be replaced with `public Schema build(final Comparator<Entry> order) {`